platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -187,7 +187,9 @@ fun <T : Any> Project.the(extensionType: KClass<T>): T =
  * Creates a [Task] with the given [name] and [type], configures it with the given [configuration] action,
  * and adds it to this project tasks container.
  */
+@Deprecated("Use tasks.register instead", ReplaceWith("tasks.register<type>(name, configuration)"))
```

Comment:
Does this suggest replacing:
```
task<Foo>("foo") { }
```
with
```
tasks.register<Foo>("foo", { })
```

Can it be made to suggest:
```
tasks.register<Foo>("foo") { }
```

Reply:
Unfortunately, no. There seems to be a limitation in the way `ReplaceWith` work.
I tried both forms, and they both give a subpar result:
`tasks.register<type>(name) { configuration }`
results in:
```
tasks.register<Copy>("foo16") {
    fun Copy.() {
        doLast { }
    }
}
```
with the whole `fun ...` highlighted in the IDE as an unused expression

while `tasks.register<type>(name, configuration)`
results in:
```
tasks.register<Copy>("foo16", fun Copy.() {
        doLast { }
})
```
which does not have any highlighted code. Deleting the not needed `fun Copy.()` then offers a fix to move the block outside of the parenthesis.

So of both, I prefer the second one.

note that I will also report it to JB as Kotlin could do better.